File: 0015-3sum/0015-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Set<List<Integer>> set = new HashSet<>();

        Arrays.sort(nums);

        for(int i=0; i < nums.length; i++){
            Set<Integer> seen = new HashSet<>();
            for(int j=i+1; j < nums.length; j++){
                int third = - (nums[i] + nums[j]);
                if(seen.contains(third)){
                    set.add(Arrays.asList(nums[i],third,nums[j]));
                }
                seen.add(nums[j]);
            }
        }

        return new ArrayList<>(set);
    }
}

File: 0015-3sum/0015-3sum.py
class Solution:
    def threeSum(self, nums: list[int]) -> list[list[int]]:
        _set = set()
        nums.sort()

        for i in range(len(nums)):
            seen = set()
            for j in range(i+1, len(nums)):
                comp = -(nums[i] + nums[j])
                if comp in seen :
                    _set.add((nums[i], comp, nums[j]))
                seen.add(nums[j])
        return list(_set)